whenchanged/whenchanged.py

```diff
@@ -63,12 +64,28 @@ def run_command(self, thefile):
         if self.run_once:
             if os.path.exists(thefile) and os.path.getmtime(thefile) < self.last_run:
                 return
+
+        self.try_stop_current_process()
+
         new_command = []
         for item in self.command:
             new_command.append(item.replace('%f', thefile))
-        subprocess.call(new_command, shell=(len(new_command) == 1))
+
+        self.start_process(new_command)
+
+    def start_process(self, command):
+        is_shell_command = len(command) == 1
+        self.process = subprocess.Popen(command, shell=is_shell_command)
         self.last_run = time.time()
 
+    def try_stop_current_process(self):
+        try:
```

Comment:
I absolutely agree with you it looks very hacky, I will refactor it in a moment. I remember that I was getting weird race condition errors without this try-except block which resulted in a complete crash, however I cannot reproduce them anymore which means that I probably did something stupid :)